ci/concourse/pipelines/kf-pipeline.yml

```diff
@@ -142,3 +148,55 @@ jobs:
       params:
         path: src
         status: failure
+
+# destroys any leftover pr pipelines
+- name: pr-cleanup
+  plan:
+  - get: pr-cleanup-trigger
+    trigger: true
+  - task: cleanup
+    config:
+      platform: linux
+      image_resource:
+        type: docker-image
+        source: *ci-image-source
+      params:
+        FLY_TARGET: *fly_target
+        FLY_USER: *fly_user
+        FLY_PASSWORD: *fly_password
+        FLY_URL: *fly_url
+        FLY_TEAM: *fly_team
+        GITHUB_REPO: *github_repo
+      run:
+        dir: ci-pipelines-src
+        path: sh
+        args:
+        - -exc
+        - |
+          # Download the fly executable for the given concourse server
+          wget "${FLY_URL}/api/v1/cli?arch=amd64&platform=linux" -O fly
+          chmod +x ./fly
+
+          # Login
+          set +x
+          echo fly login...
+          ./fly login\
+            -t "$FLY_TARGET" \
+            -c "$FLY_URL" \
+            -n "$FLY_TEAM" \
+            -u "$FLY_USER" \
+            -p "$FLY_PASSWORD"
+          set -x
+
+          prs=$(mktemp)
+          curl -L "https://api.github.com/repos/${GITHUB_REPO}/pulls?state=open" | jq .[].number | sort | uniq > $prs
```

Comment:
Do we have to worry about rate limiting when hitting the github APIs? I realize we aren't hitting it often, but will our requests be lumped into other requests coming from GCP data-centers as we aren't providing a token?

Reply:
Sure, I'll add the request token you already created